api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,41 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  /**
+   * Try to get the 'X-Access-Token' from the request headers.
+   * If no token is provided,
+   *   throw a 401 error with the message 'authentication required'
+   * Use jwt.verify() to verify the authenticity of the token and extract its payload
+   * Assign the extracted payload to the user property of the req object.
+   * Call next() (with no arguments) to let Express know to advance to its next route or middleware.
+   */
+
+  try {
+    const xAccessToken = req.headers['x-access-token'];
+    if (!xAccessToken) {
+      throw new ClientError(401, 'authentication required');
+    }
+
+    try {
+      const payload = jwt.verify(xAccessToken, process.env.TOKEN_SECRET);
+      req.user = payload;
+    } catch (err) {
+      next(err);
+    }
+    next();
+
+  } catch (err) {
+    next(err);
+  }
```

Comment:
You don't need the nested `try catch` you can do:
```JAVASCRIPT
  try {
    const xAccessToken = req.headers['x-access-token'];
    if (!xAccessToken) {
      throw new ClientError(401, 'authentication required');
    }

    const payload = jwt.verify(xAccessToken, process.env.TOKEN_SECRET);
    req.user = payload;
    next();
  } catch (err) {
    next(err);
  }
```

This will work the same way, without the extra `try catch`. And actually because the `errorMiddleware` is setup to handle specific errors from `jwt` you can do this without any `try catch` and it will still work correctly and catch the errors.